gtfu/src/test/java/gtfu/test/ServerTest.java

```diff
@@ -120,22 +121,46 @@ private static boolean checkForServiceAlertFeed(String agency) throws Exception
         }
     }
 
+    // Checks whether the provided agency has a position updated logged to the DB within the last 30 seconds.
+    // Note that the update is posted by running `NODE_PATH=../node/node_modules node post-position-update.js` from server/tests
+    private static boolean checkDBForPosUpdate(String agency) throws Exception {
+        String[] propertyNames = {"agency-id"};
+        int searchWindowSecs = 30;
+
+        DB db = new DB();
+        long nowSecs = Util.now() / 1000;
+        long queryStartTime = nowSecs - searchWindowSecs;
+        long queryEndTime = nowSecs;
+        List<String> results = db.fetch(queryStartTime, queryEndTime, "position", propertyNames);
+
+        for (int i=0; i < results.size(); i++){
+            if(results.get(i).equals(agency)){
```

Comment:
The assumption here is that no one else is submitting updates as agency `test`, which is of course not guaranteed. Agency name `test` is currently hardcoded in post-position-updates.js. It would be nice if a) there was a way to pass an arg when invoking this (whether it's agency id or something else) and then b) pass a randomly generated value and check for exactly that value in the code above.

Reply:
I misread the test code. Agency id is set to `pr-test`. not `test`. Could still come from somewhere else, but much less likely. So, feel free to disregard.

Reply:
I think the approach you described would be more watertight, and that existing will work well for now